firebase_admin/_sseclient.py

```diff
@@ -0,0 +1,198 @@
+        #  self.resp.raw._fp.fp.raw._sock.close()
+
+
+    def _connect(self):
+        """connects to the server using requests"""
+        if self.should_connect:
+            success = False
+            while not success:
+                if self.last_id:
+                    self.requests_kwargs['headers']['Last-Event-ID'] = self.last_id
+                headers = self.build_headers()
+                self.requests_kwargs['headers'].update(headers)
+                # Use session if set.  Otherwise fall back to requests module.
+                self.requester = self.session or requests
+                self.resp = self.requester.get(self.url, stream=True, **self.requests_kwargs)
+
+                self.resp_iterator = self.resp.iter_content(decode_unicode=True)
+
+                # TODO: Ensure we're handling redirects.  Might also stick the 'origin'
+                # attribute on Events like the Javascript spec requires.
+                self.resp.raise_for_status()
+                success = True
+        else:
+            raise StopIteration()
+
+    def _event_complete(self):
+        return re.search(end_of_field, self.buf) is not None
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        while not self._event_complete():
+            try:
+                nextchar = next(self.resp_iterator)
+                self.buf += nextchar
+            except (StopIteration, requests.RequestException):
+                time.sleep(self.retry / 1000.0)
+                self._connect()
+
+                # The SSE spec only supports resuming from a whole message, so
+                # if we have half a message we should throw it out.
+                head, sep, tail = self.buf.rpartition('\n')
+                self.buf = head + sep
+                continue
+
+        split = re.split(end_of_field, self.buf)
+        head = split[0]
+        tail = "".join(split[1:])
+
+        self.buf = tail
+        msg = Event.parse(head)
+
+        if msg.data == "credential is no longer valid":
+            self._connect()
+            return None
+
+        if msg.data == 'null':
+            return None
+
+        # If the server requests a specific retry delay, we need to honor it.
+        if msg.retry:
+            self.retry = msg.retry
+
+        # last_id should only be set if included in the message.  It's not
+        # forgotten if a message omits it.
+        if msg.event_id:
+            self.last_id = msg.event_id
+
+        return msg
+
+    if six.PY2:
+        next = __next__
+
+
+class Event(object):
+    """Event class to handle the events fired by SSE"""
+
+    sse_line_pattern = re.compile('(?P<name>[^:]*):?( ?(?P<value>.*))?')
+
+    def __init__(self, data='', event='message', event_id=None, retry=None):
+        self.data = data
+        self.event = event
+        self.event_id = event_id
+        self.retry = retry
+
+    def dump(self):
```

Comment:
I think this whole method can be removed.

Reply:
Method Removed, and it's accompanying test in `test/test_sseclient.py` removed.